src/v/raft/consensus.h

```diff
@@ -527,6 +532,7 @@ class consensus {
     raft::group_id _group;
     timeout_jitter _jit;
     storage::log _log;
+    ss::lw_shared_ptr<offset_translator> _offset_translator;
```

Comment:
@mmaslankaprv I decided to place this on the raft layer instead of somewhere lower because the integration with raft is really straightforward, especially around hydrating the snapshot at startup - we can't start the offset translator using only log, we need info from the snapshot. I'm open to discussion though.

Reply:
do we need this to be a pointer with shared semantics ?

Reply:
One place where I thought that might be necessary is this: https://github.com/vectorizedio/redpanda/pull/2493/files#diff-45ca508d3c9cac1b0ebcbbd1b9e24c4b3b836c576ccadd0a67e6d87a80ae6188R82

It is not immediately obvious to me that offset translator always lives longer than the reader. If that's the case, we can use ordinary references.

Reply:
I think this is good, i.e. when we destroy partition it may happen that it will be deleted before the read end. Let's keep it as shared pointer.